ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -108,11 +106,13 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, NetworkMetr
       List<NetworkSend> sends = prepareSends(responseInfoList);
       selector.poll(pollTimeoutMs, sends);
       handleSelectorEvents(responseInfoList);
-      return responseInfoList;
+    } catch (Exception e) {
+      networkMetrics.networkClientException.inc();
```

Comment:
Also not sure if this is a good place to catch these errors. I would say that the `NetworkClient::sendAndPoll()` should throw and the caller can handle the problems.